stingray/crosscorrelation.py

```diff
@@ -251,9 +245,20 @@ def cal_timeshift(self, dt=1.0):
                     self._make_corr(self.lc1, self.lc2)
 
         self.n = len(self.corr)
-        dur = int(self.n / 2)
-        # Correlation against all possible lags, positive as well as negative lags are stored
-        x_lags = np.linspace(-dur, dur, self.n)
+        
```

Comment:
@matteobachetti As you suggested, I have moved these operations to `cal_timeshift` internal method.

Reply:
@matteobachetti 
All tests are failing for one test case - `test_cross_correlation_with_unequal_lc()` in test_crosscorrelation.py. The reason is - time_lags obtained by `scipy.signal.correlation_lag()` are diffrent ([-1, 0, 1, 2, 3]) from of time_lags given in the test case ([-2, -1, 0, 1, 2]). Since the crosscorealtion is calculated using `scipy.signal.correlate()` it makes more sense to change the time_lags in the test case accordingly. Should I make the changes in the test case?